BibBuddy/app/src/main/java/de/bibbuddy/BookFragment.java

```diff
@@ -218,56 +271,56 @@ public void onRequestPermissionsResult(int requestCode,
     }
   }
 
-  private void createBibFile(String folderName, String fileName) {
+  private void bibExportShelf(String fileName) {
+    String rootPathStr = Environment.getExternalStorageDirectory() + File.separator
+        + folderName + File.separator + fileName + fileTypeBib;
     try {
-      String rootPath = Environment.getExternalStorageDirectory() + "/" + folderName + "/";
-      File root = new File(rootPath);
-
-      if (!root.exists()) {
-        root.mkdirs();
-      }
-
-      File file = new File(rootPath + fileName + ".bib");
+      File file = new File(rootPathStr);
       if (file.exists()) {
         file.delete();
       }
-      file.createNewFile();
 
+      file.createNewFile();
       FileOutputStream out = new FileOutputStream(file);
       out.flush();
       out.close();
 
-      Toast.makeText(getContext(), getString(R.string.exported_file_stored_in) + '\n'
-          + "/" + folderName + "/" + fileName + ".bib", Toast.LENGTH_LONG).show();
+      writeBibFile(shelfName);
 
     } catch (Exception e) {
       e.printStackTrace();
     }
+
+    Toast.makeText(getContext(),
+        getString(R.string.exported_file_stored_in) + '\n'
+            + File.separator + folderName + File.separator + fileName
+            + fileTypeBib, Toast.LENGTH_LONG).show();
   }
 
-  private void retrieveBibContent() {
+  private String retrieveBibContent() {
 
-    DatabaseHelper dbHelper = new DatabaseHelper(getContext());
-    BookDao bd = new BookDao(dbHelper);
-    NoteDao nd = new NoteDao(dbHelper);
+    dbHelper = new DatabaseHelper(getContext());
+    bookDao = new BookDao(dbHelper);
+    NoteDao noteDao = new NoteDao(dbHelper);
 
-    List<Long> bookIdsCurrShelf = bd.getAllBookIdsForShelf(shelfId);
+    List<Long> bookIdsCurrShelf = bookDao.getAllBookIdsForShelf(shelfId);
     List<Author> authorsCurrBook = new ArrayList<>();
     List<Long> notesCurrBook = new ArrayList<>();
     String allNotesCurrBook;
     String authorNamesCurrBook;
+    String bibShelfContent = "";
 
     //for each book in the current shelf
     for (int i = 0; i < bookIdsCurrShelf.size(); i++) {
 
       Book currBook;
```

Comment:
currentBook is a better name
```suggestion
      Book currentBook;
```

Reply:
If I change currBook to currentBook, then this means that I should change all variable names where I use "curr" for current. In case for the variable "bookIdsCurrShelf" it should then  be "bookIdsCurrentShelf", which I think is too long and "curr" is self-explained that it stands for "current".

Reply:
For me curr is not self-explained. Until now Sarah and I always used current in the names instead of curr.